components/brave_ads/browser/BUILD.gn

```diff
@@ -16,6 +16,7 @@ source_set("browser") {
   deps = [
     "//base",
     "//brave/components/brave_ads/common",
+    "//brave/components/brave_rewards/common",
```

Comment:
this should have brave_rewards/browser too for the RewardsServiceFactory

Reply:
done